src/services/fluence-aurora-connector/modules/fluence_aurora_connector/src/main.rs

```diff
@@ -31,422 +41,78 @@ enum Error {
+        return MultipleDealsChanged::empty();
     }
 
+    let batch = deal_changed_req_batch(&deals);
+    let responses = send_jsonrpc_batch::<GetLogsReq, Vec<Log>>(api_endpoint, batch);
+    let responses = match responses {
+        Err(err) => return MultipleDealsChanged::error(err.to_string()),
+        Ok(r) => r,
+    };
+
     let mut updated_deals = Vec::new();
-    let reqs = deals
-        .iter()
-        .enumerate()
-        .map(|(idx, deal)| {
-            let to_block = get_to_block(&deal.from_block);
-            let address = format!("0x{}", deal.deal_info.deal_id);
-            let req = GetLogsReq {
-                address,
-                topics: vec![DealChangedData::topic()],
-                from_block: deal.from_block.clone(),
-                to_block,
-            };
-            req.to_jsonrpc(idx as u32)
-        })
-        .collect::<Vec<_>>();
-    let result = get_logs_batch(api_endpoint, reqs);
-    match result {
-        Err(err) => {
-            return DealsUpdatedBatchResult::error(err.to_string());
-        }
-        Ok(results) => {
-            for (deal, result) in std::iter::zip(deals, results) {
-                let to_block = get_to_block(&deal.from_block);
-                match result.get_result() {
-                    Err(err) => {
-                        let result = DealUpdatedBatchResult::error(
-                            to_block,
-                            deal.deal_info,
-                            err.to_string(),
-                        );
-                        updated_deals.push(result);
-                    }
-                    Ok(result) => {
-                        let latest_update: Option<DealChanged> = try {
-                            let update = result.into_iter().filter(|deal| !deal.removed).last()?;
-                            parse_deal::<DealChangedData, DealChanged>(update)?
-                        };
-                        if let Some(update) = latest_update {
-                            let result =
-                                DealUpdatedBatchResult::ok(to_block, deal.deal_info, update);
-                            updated_deals.push(result);
-                        }
-                    }
+
+    for (deal, result) in zip(deals, responses) {
+        let to_block = default_right_boundary(&deal.left_boundary);
+        match result.get_result() {
+            Err(err) => {
+                let result = DealChangedResult::error(to_block, deal.deal_info, err.to_string());
+                updated_deals.push(result);
+            }
+            Ok(result) => {
+                let last_log = result.into_iter().filter(|deal| !deal.removed).last();
+                let change = last_log.and_then(parse_deal_changed);
+                if let Some(change) = change {
+                    let result = DealChangedResult::ok(to_block, deal.deal_info, change);
+                    updated_deals.push(result);
```

Comment:
```suggestion
                let latest_changed_deal: Option<_> = try {
                    let last_log = result.into_iter().filter(|deal| !deal.removed).last()?;
                    parse_deal_changed(last_log)?
                };
                if let Some(latest_changed_deal) = latest_changed_deal {
                    updated_deals.push(DealChangedResult::ok(to_block, deal.deal_info, latest_changed_deal));
```

mb?

Reply:
I don't see how that's simpler or better